src/lpm.c

```diff
@@ -711,110 +718,233 @@ static int lpm_extract(lua_State* L) {
           }
+          
           char buffer[8192];
           FILE* file = fopen(target, "wb");
           if (!file) {
             mtar_close(&tar);
             return luaL_error(L, "can't extract tar archive file %s, can't create file %s: %s", src, target, strerror(errno));
           }
+          
           if (chmod(target, h.mode))
             return luaL_error(L, "can't extract tar archive file %s, can't chmod file %s: %s", src, target, strerror(errno));
+            
           int remaining = h.size;
           while (remaining > 0) {
             int read_size = remaining < sizeof(buffer) ? remaining : sizeof(buffer);
+            
             if (mtar_read_data(&tar, buffer, read_size) != MTAR_ESUCCESS) {
               fclose(file);
               mtar_close(&tar);
               return luaL_error(L, "can't write file %s: %s", target, strerror(errno));
             }
+            
             fwrite(buffer, sizeof(char), read_size, file);
             remaining -= read_size;
           }
+          
           fclose(file);
         }
+        
+        // !!! NOT REALLY TESTED !!!
+        else if (h.type == MTAR_TEHR || h.type == MTAR_TEHRA) {
+          mtar_header_t *h_to_change;
+          if (h.type == MTAR_TEHR)
+            h_to_change = &before_h;
+          else
+            h_to_change = &allways_h;
+          
+          char buffer[4096];
+          char curent_read[8192]; // If a line is more than 8192 char long, will not work!
+          char last_read[4096] = {0};
+          int remaining = h.size;
+          
+          has_ext_before = 1;
+          
+          while (remaining > 0) {
+            int read_size = remaining < sizeof(buffer) ? remaining : sizeof(buffer);
+            remaining -= read_size;
+            
+            if (mtar_read_data(&tar, buffer, read_size) != MTAR_ESUCCESS) {
+              mtar_close(&tar);
+              return luaL_error(L, "Error while reading extended: %s", strerror(errno));
+            }
+            
+            strcpy(curent_read, last_read);
+            curent_read[strlen(last_read)] = '\0';
+            strcat(curent_read, buffer);
+            curent_read[strlen(last_read) + read_size] = '\0';
+            
+            char *n_line_ptr = NULL;
+            char **l_line_ptr = NULL;
+            char *line = strtok_r(curent_read, "\n", &n_line_ptr);
+            
+            while (line != NULL) {
+              char *in_line_ptr = NULL;
+              strtok_r(line, " ", &in_line_ptr);
+              char *header_key = strtok_r(line, "=", &in_line_ptr);
+              char *header_val = strtok_r(line, "=", &in_line_ptr);
+              
+              if (header_key == "path")      strcpy(h_to_change->name, header_val);
+              if (header_key == "linkpath")  strcpy(h_to_change->linkname, header_val);
+                // possibility to add more later
+              
+              printf("%s\n", line);
```

Comment:
I also assume this line is a debug statement that got left in?

Reply:
yep, it should be gone by now